lib/karafka/contracts/config.rb

```diff
@@ -9,71 +9,90 @@ module Contracts
     #   validated per each route (topic + consumer_group) because they can be overwritten,
     #   so we validate all of that once all the routes are defined and ready.
     class Config < Base
-      params do
-        # License validity happens in the licenser. Here we do only the simple consistency checks
-        required(:license).schema do
-          required(:token) { bool? | str? }
-          required(:entity) { str? }
-          required(:expires_on) { date? }
+      configure do |config|
+        config.error_messages = YAML.safe_load(
+          File.read(
+            File.join(Karafka.gem_root, 'config', 'errors.yml')
+          )
+        ).fetch('en').fetch('validations').fetch('config')
+      end
+
+      # License validity happens in the licenser. Here we do only the simple consistency checks
+      nested(:license) do
+        required(:token) { |token| [true, false].include?(token) || token.is_a?(String) }
```

Comment:
We could simplify this by using `val` in the block. Based on the argument's name, we already know what we are operating on.